data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostAttesterSlashing.java

```diff
@@ -19,33 +19,53 @@
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_BEACON;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.HTTP_ERROR_RESPONSE_TYPE;
 
-import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.javalin.http.Context;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.jetbrains.annotations.NotNull;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.NodeDataProvider;
-import tech.pegasys.teku.api.schema.AttesterSlashing;
-import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
-import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.spec.datastructures.operations.IndexedAttestation;
 import tech.pegasys.teku.statetransition.validation.InternalValidationResult;
 import tech.pegasys.teku.statetransition.validation.ValidationResultCode;
 
-public class PostAttesterSlashing extends AbstractHandler {
+public class PostAttesterSlashing extends MigratingEndpointAdapter {
   public static final String ROUTE = "/eth/v1/beacon/pool/attester_slashings";
   private final NodeDataProvider nodeDataProvider;
 
-  public PostAttesterSlashing(final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    this(dataProvider.getNodeDataProvider(), jsonProvider);
+  public PostAttesterSlashing(final DataProvider dataProvider, final Spec spec) {
+    this(dataProvider.getNodeDataProvider(), spec);
   }
 
-  public PostAttesterSlashing(final NodeDataProvider provider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
+  public PostAttesterSlashing(final NodeDataProvider provider, final Spec spec) {
+    super(
+        EndpointMetadata.post(ROUTE)
+            .operationId("postAttesterSlashing")
+            .summary("Submit attester slashing object")
+            .description(
+                "Submit signed attestations to the beacon node to be validated and submitted if valid.\n\n"
+                    + "This endpoint does not protected against slashing.")
```

Comment:
not sure why that second part is there, delete second line... `this endpoint does not protected` has issues, but im struggling to understand why a post slashing endpoint would be protected against slashing...

Reply:
The entire description appears to be copied from the send signed attestation endpoint - it's talking about attestations not slashing.